docs/Writing_easyconfig_files.rst

```diff
@@ -506,12 +506,86 @@ which is equivalent to::
 
   dependencies = [('PnMPI', '1.2.0', '', True)]
 
+
 Using external modules as dependencies
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Since EasyBuild v2.1, specifying modules that are not provided via EasyBuild as dependencies is also supported.
 See :ref:`using_external_modules` for more information.
 
+
+Module Extensions
+~~~~~~~~~~~~~~~~~
+
+Besides dependencies, which are found outside the module being built but are part of the site's EasyBuild installation,
+it is also possible to incorporate extensions to the module within the build.  This is done via the ``exts_list`` array.
+
+Each entry in ``exts_list`` is a three-component tuple, with the name and
+version number, and a dictionary of source options:
+
+.. code:: python
+
+  exts_list = [
+      ('name', 'version', { 'option':'value', 'option':'value' })
+  ]
+
+The latter consist of most of the allowed keys as in the dictionary structure of ``sources`` (see
+:ref:`_common_easyconfig_param_sources_alt`).  Among those options, the following exceptions and
+special cases should be noted:
+
+* **nosource**: If set ``True``, no download will be done
+* **filename** and **download_filename** are ignored; use ``'source_tmpl'``
+* **source_tmpl**: Template string for the file to be downloaded
+  * default is ``'%(name)s-%(version)s.tar.gz'``
+  * ``%(name)s`` and ``%(version)s`` come from the ``exts_list`` entry (above)
+* **git_config**: Dictionary specifying download from a Git repository
+* **start_dir**: If not set, will be derived; the EasyConfigs value will not be used
+
+.. code:: python
+
+  exts_list = [
+      ('llvmlite', '0.26.0', {
+          'source_urls': ['https://pypi.python.org/packages/source/l/llvmlite/'],
+          'patches': ['llvmlite-0.26.0_fix-ffi-Makefile.patch'],
+          'checksums': [
+              '13e84fe6ebb0667233074b429fd44955f309dead3161ec89d9169145dbad2ebf',    # llvmlite-0.26.0.tar.gz
+              '40e6fe6de48709b45daebf8082f65ac26f73a4afdf58fc1e8099b97c575fecae',    # llvmlite-0.26.0_fix-ffi-Makefile.patch
+          ],
+      }),
+      ('singledispatch', '3.4.0.3', {
+          'source_urls': ['https://pypi.python.org/packages/source/s/singledispatch/'],
+          'checksums': ['5b06af87df13818d14f08a028e42f566640aef80805c3b50c5056b086e3c2b9c'],
+      }),
+      (name, version, {
+          'source_urls': ['https://pypi.python.org/packages/source/n/numba/'],
+          'checksums': ['c62121b2d384d8b4d244ef26c1cf8bb5cb819278a80b893bf41918ad6d391258'],
+      }),
+  ]
+
+That third instance uses the ``name`` and ``version`` variables defined in the EasyConfigs file.
+
+Since EasyBuild v4.2.1, the ``git_config`` dictionary (see :ref:`downloading-from-a-git-repository`)
+may be included in the ``exts_list`` source definition.  For example, to download Git sources from a
+private repository and convert them to a tar-ball for installation:
+
+.. code:: python
+
+  exts_defaultclass = 'PythonPackage'
+  exts_list = [
+      ('pyCAP', 'master', {
```

Comment:
It's a detail, but I'd personally prefer to see a different tag as the `version` here, the example is going to give a non-reproducible build. How about:
```
    ('pyCAP', '0.1', {
          'source_tmpl': '%(name)s-%(version)s.tar.gz',
          'git_config': {
              'url': 'ssh://nero.stanford.edu/data/git/Analysis',
              'repo_name': 'pyCAP',
              'tag': 'v%(version)s',
          }
      }),
```

Reply:
Point taken.  I can give it as an example, but that particular (private!) package doesn't have tags (we're slowly trying to migrate away from using it at all).